### PR TITLE
Dependency updates, Alpine-based image, CI builds, cross compilation

### PR DESCRIPTION
**Pull request by Gelbpunkt**
This does a few things:

- The Dockerfile is now based on Alpine Linux for the build part
- The build takes 3 arguments for the architecture to build for, this could potentially be scripted to only take one argument, but for now I made it manual as everything uses different wordings for the architectures (see the comments in the file)
- I added a Github action that builds the image for x86_64 and aarch64 via cross-compilation (way faster than QEMU) and pushes it as a combined manifest to Dockerhub
- I updated the dependencies, removed openssl entirely as it causes issues in Alpine and fixed clippy lints

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,12 +5,12 @@ name = "twilight-http-proxy"
...
-twilight-http = { branch = "trunk", git = "https://github.com/twilight-rs/twilight.git" }
+twilight-http = "0.1"
 hyper = "0.13"
 http = "0.2"
 tracing = "0.1"
 tracing-subscriber = { version = "0.2", features = ["fmt", "registry"] }
 tracing-log = "0.1"
-reqwest = "0.10"
-snafu = "0.5"
+reqwest = { version = "0.10", default-features = false }
```
Comment (Erk-) [inline Cargo.toml]: You should have a TLS method here, it only works currently because rustls-TLS is enabled elsewhere in the tree

**Review by Erk- — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (Gelbpunkt) [reply Cargo.toml]: Done :+1:

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -14,18 +14,22 @@ jobs:
             echo "{\"experimental\": \"enabled\"}" > ~/.docker/config.json
       - name: Login to DockerHub
         uses: docker/login-action@v1
+        if: ${{ secrets.DOCKERHUB_USERNAME }}
         with:
-          username: gelbpunkt
+          username: if: ${{ secrets.DOCKERHUB_USERNAME }}
```
Comment (nickelc) [inline .github/workflows/publish.yml]:
```suggestion
          username: ${{ secrets.DOCKERHUB_USERNAME }}
```

Comment (Gelbpunkt) [reply .github/workflows/publish.yml]: Was fixed :)

**Comment by Gelbpunkt** [comment]
It now only logs in and pushes if the secrets for username and password are available. Means that in forks and PRs, builds will still be done, but not published. All references to my Dockerhub account are also gone

**Review by Erk- — APPROVED** [review]

**Comment by BlackHoleFox** [comment]
I'm not a huge fan of MUSL but Alpine Linux is appealing to a certain extent. How large do the final images come out to be?

**Comment by Gelbpunkt** [comment]
> I'm not a huge fan of MUSL but Alpine Linux is appealing to a certain extent. How large do the final images come out to be?

4.9MB for aarch64 and 5.1MB for x86_64 (that's the compressed size dockerhub reports)

**Review by spring4175 — APPROVED** [review]
Actions isn't my thing but the dockerfile and source changes looks fine

**Pull request merged**